Shared/Realm.Sync.Shared/Exceptions/ErrorCodeHelper.cs

```diff
@@ -0,0 +1,43 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+
+namespace Realms.Sync
+{
+    internal static class ErrorCodeHelper
+    {
+        public static ErrorCode? GetErrorCode(int? statusCode)
+        {
+            switch (statusCode)
+            {
+                case null:
+                case 0:
+                    return null;
+                default:
+                    var errorCode = (ErrorCode)statusCode.Value;
+                    if (!Enum.IsDefined(typeof(ErrorCode), errorCode))
+                    {
+                        errorCode = ErrorCode.Unknown;
```

Comment:
StatusMessage is populated by the server - it should have the correct error description set.